Storage/Site.swift

```diff
@@ -74,6 +74,7 @@ public class Site : Identifiable {
      // Sites may have multiple favicons. We'll return the largest.
     public var icon: Favicon?
     public var latestVisit: Visit?
+    public var didReplaceSuggestedSite: Bool = false
```

Comment:
It doesn't seem like a good idea for deleted top site state to live here, but I'll take a look again once I better understand what you're trying to do.